### PR TITLE
Make CTT dependency optional.

### PR DESCRIPTION
**Pull request by obilaniu**
# Description

Separate out ctt and speedrun into an optional feature, with its own dependencies.
The package with its new feature can be installed using

    pip install 'covid19sim[ctt]'

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by plstcharles — APPROVED** [review]

**Comment by pg2455** [comment]
@satyaog  - Do we need to change any tests with regards to this change?

**Comment by satyaog** [comment]
@pg2455 yes a small of change will be needed in the setup of the CI

File: README.md [hunk 1]
```diff
@@ -16,11 +16,18 @@ We plan to update the simulator as more and more about COVID-19 will be known.
 
 
 ## Dependencies
-Following `python` packages are required (python>=3.7)
+To install the simulator without `ctt`
 ```
-pip install -r requirements.txt
+pip install covid19sim
```
Comment (satyaog) [inline README.md]: I think for now it should read `pip install .`

File: README.md [hunk 2]
```diff
@@ -16,11 +16,18 @@ We plan to update the simulator as more and more about COVID-19 will be known.
...
 ```
 
+To install the simulator with [`ctt`](https://github.com/nasimrahaman/ctt)
+```
+pip install 'covid19sim[ctt]'
```
Comment (satyaog) [inline README.md]: I think for now it should read `pip install '.[ctt]'`.

File: README.md (hunk 1 above)
Comment (pg2455) [reply README.md]: Ok! thanks for pointing that out. fixed it!

**Comment by pg2455** [comment]
@satyaog  - Can we make those changes with this PR itself?

**Comment by satyaog** [comment]
@pg2455 , one way or the other. This PR can be merged then I’ll rebase on the new develop or merge mine then I’ll update the CI  script in this one

**Comment by pg2455** [comment]
@satyaog  - there is a conflict here now.

**Review by satyaog — APPROVED** [review]

**Comment by obilaniu** [comment]
@pg2455 Go for merge then?

**Comment by satyaog** [comment]
Btw, I added the same tweak to the ctt project to decouple the tensorflow depedency just before this got merged. The python 3.8 test is still running but should be fine

**Pull request merged**